internal/chezmoi/realsystem.go

```diff
@@ -100,10 +100,20 @@ func (s *RealSystem) RunScript(scriptname RelPath, dir AbsPath, data []byte) (er
 	// Run the temporary script file.
 	//nolint:gosec
 	cmd := exec.Command(f.Name())
-	cmd.Dir = string(dir)
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+
+	// The requested working dir may not actually exist yet, if this is a
+	// `before_` script.  In that case, look through the parent hierarchy till
+	// we find one.
+	for workDir := dir; string(dir) != "."; workDir = workDir.Dir() {
```

Comment:
Is this an idiomatic way to do this in Go?  I'm usually a `{Java,Type}Script` person, so sometimes I'm not sure :wink:

Reply:
Yup, this totally works. Also, as `dir` is not used again after this loop, it would be equally good to mutate the `dir` variable in a while loop.